aea/components/loader.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains utilities for loading components."""
+import re
+from typing import Dict, Type
+
+from aea.configurations.base import ComponentConfiguration, ComponentType
+from aea.configurations.components import Component
+from aea.connections.base import Connection
+from aea.contracts.base import Contract
+from aea.exceptions import AEAPackageLoadingError
+from aea.protocols.base import Protocol
+from aea.skills.base import Skill
+
+
+def component_type_to_class(component_type: ComponentType) -> Type[Component]:
+    type_to_class: Dict[ComponentType, Type[Component]] = {
+        ComponentType.PROTOCOL: Protocol,
+        ComponentType.CONTRACT: Contract,
+        ComponentType.CONNECTION: Connection,
+        ComponentType.SKILL: Skill,
+    }
+    return type_to_class[component_type]
+
+
+def load_component_from_dir(component_type: ComponentType, directory: str) -> Component:
+    """
+    Load a component from a directory.
+
+    :param component_type: the component type.
+    :param directory: the directory.
+    :return: the component instance.
+    """
+    component_class = component_type_to_class(component_type)
+    return component_class.from_dir(directory)  # type: ignore
+
+
+def load_component_from_config(  # type: ignore
```

Comment:
This is used by the AEA builder as a single entry point to load components.

However, it isn't perfect. It has only been applied to protocols and contracts because:
- the loading of a skill requires an agent context.
- the loading of a connection works differently. It requires to load a connection class first, and then we can call the `from_config` method.

Reply:
I think above you use it for skills? is this message out of date?

Reply:
Ah I see what you mean now..